src/twisted/trial/test/test_deferred.py

```diff
@@ -220,7 +229,29 @@ def test_errorPropagation(self):
         result = self.runTest("test_errorPropagation")
         self.assertFalse(result.wasSuccessful())
         self.assertEqual(result.testsRun, 1)
-        self._wasTimeout(detests.TimeoutTests.timedOut)
+        error = detests.TimeoutTests.timedOut
+        self.assertEqual(error.check(defer.CancelledError), defer.CancelledError)
```

Comment:
This is *technically* a breaking change. Rather than throwing a TimeoutError into the test code, trial now uses the standard deferred timeout mechanism which throws a CancelledError instead.

However this only occurs at the point of failing a trial user's test anyway, and the old errback approach means this exception couldn't be caught without `.crash`ing the reactor which puts this change well outside the realm of supported behaviour

Reply:
I don't think that's true. It's a change, but canceling a `Deferred` is never a breaking change - your caller could always do it if the `Deferred` is async.